packages/kotti-ui/source/kotti-filter/types.ts

```diff
@@ -52,28 +52,35 @@ export namespace KottiFilter {
 		label: string
 	} & (
 		| {
-				type: FilterType.SEARCH
+				operations: Operation.String[]
```

Comment:
i think we fell in the same trap with a `Value` type for `KtField*Select`s

> if you support operations to be an array of all String Operations, and the user-app decides to use a subset of those, then you type your column on the user-app's side to be `Column`, it will basically make TS think you support the full array of operations

so, you basically might want to introduce a generic that by default extends all the operations, depending on the type or accepts the subset of operations (see this issue https://github.com/3YOURMIND/kotti/issues/377)

Reply:
except if I'm not missing something, your generics here would be two (`T`, `V`, for example) where `T` is the `type` and `V` depends on `T` and has a default value that = all of `T`'s supported OPs and always extends it `Operation<T>`